program/src/instructions/shared_accounts.rs

```diff
@@ -629,6 +652,20 @@ impl<'info> ValidateAsset<'info> for T22Shared<'info> {
 impl<'info> ValidateAsset<'info> for MplxShared<'info> {
     fn validate_asset(&self) -> Result<AmmAsset> {
         let metadata = assert_decode_metadata(&self.mint.key(), &self.metadata)?;
+
+        // Verify edition is valid, this should be a Master Edition or Edition.
+        let key = self.edition.try_borrow_data()?[0];
+
+        match key {
+            k if k == MplKey::MasterEditionV1 as u8 || k == MplKey::MasterEditionV2 as u8 => {
+                assert_decode_master_edition(&self.edition)?;
+            }
+            k if k == MplKey::EditionV1 as u8 => {
+                assert_decode_edition(&self.edition)?;
+            }
+            _ => return Err(ErrorCode::InvalidEdition.into()),
+        }
```

Comment:
Are there any cases where we don't have an edition?

Reply:
Supporting fungibles should be a deliberate change, so for now all `Legacy` or `pNFT` Metaplex assets should have an edition.